app/src/server/api/routers/datasetEvals.router.ts

```diff
@@ -469,4 +470,51 @@ export const datasetEvalsRouter = createTRPCRouter({
         originalOutput: datasetEvalDatasetEntry?.datasetEntry?.output,
       };
     }),
+
+  results: protectedProcedure
+    .input(z.object({ datasetEvalId: z.string() }))
+    .query(async ({ input, ctx }) => {
+      const datasetEval = await prisma.datasetEval.findUniqueOrThrow({
+        where: { id: input.datasetEvalId },
+        include: { dataset: { select: { projectId: true } } },
+      });
+      await requireCanViewProject(datasetEval.dataset.projectId, ctx);
+
+      const baseQuery = kysely
+        .selectFrom("DatasetEvalOutputSource as deos")
+        .innerJoin("DatasetEvalResult as der", "der.datasetEvalOutputSourceId", "deos.id")
+        .innerJoin("DatasetEvalDatasetEntry as dede", "dede.id", "der.datasetEvalDatasetEntryId")
+        .innerJoin("DatasetEntry as de", "de.id", "dede.datasetEntryId")
+        .leftJoin("FineTune as ft", (join) => join.onRef(sql`ft.id::text`, "=", "deos.modelId"))
+        .where("de.outdated", "=", false)
+        .where("dede.datasetEvalId", "=", input.datasetEvalId)
+        .select((eb) => [
+          "deos.modelId as modelId1",
+          "ft.slug as slug1",
+          eb.fn.avg<number>("der.score").as("winRate"),
+          sql<number>`count(case when der.score = 1 then 1 end)::int`.as("wins"),
+          sql<number>`count(case when der.score = .5 then 1 end)::int`.as("ties"),
+          sql<number>`count(case when der.score = 0 then 1 end)::int`.as("losses"),
+        ]);
+
+      const [leaderboard, headToHead] = await Promise.all([
+        baseQuery.groupBy(["modelId1", "slug1"]).orderBy("winRate", "desc").execute(),
+
+        baseQuery
```

Comment:
I think you could take advantage of the fact that we store the comparison output source on results and skip this join to the comparison results.

Reply:
Ah yes you're probably right. I'm too tired to handle this tonight and need it in for tomorrow so won't make this change right now, but will do so next time I touch this code (or you're free to).